scripts/autoregress.py

```diff
@@ -0,0 +1,168 @@
+	if verbosity == 2:
+		logger.setLevel(logging.DEBUG)
+	elif verbosity == 0:
+		logger.setLevel(logging.ERROR)
+	elif verbosity == 1:
+		logger.setLevel(logging.INFO)
+	if args.output == "-":
+		logger.addHandler(logging.StreamHandler(sys.stdout))
+	elif args.output != None:
+		logger.addHandler(logging.FileHandler(args.output))
+	else:
+		logger.addHandler(logging.handlers.SysLogHandler("/dev/log"))
+	clean = args.clean
+	workdir = os.path.abspath(args.dir)
+	logger.debug("tmpdir = %s" % tmpdir)
+	logger.debug("workdir = %s" % workdir)
+	if os.path.exists(tmpdir):
+		logger.warn("directory %s exists, deleting contents" % tmpdir)
+		shutil.rmtree(tmpdir)
+	logger.debug("creating directory %s" % tmpdir)
+	os.mkdir(tmpdir)
+	failures = display_results(autoregress(workdir, tmpdir, verbosity,
+			clean))
+	if clean:
+		os.rmdir(tmpdir)
+	sys.exit(failures)
+
+# Log how many tests succeeded and failed and return the number of failures.
+def display_results(results):
+	logger = logging.getLogger(__name__)
+	successes = 0
+	failures = 0
+	for result in results:
+		if result == True:
+			successes += 1
+		if result == False:
+			failures += 1
+	logger.info("%i tests succeeded and %i tests failed" % (successes,
+			failures))
+	return failures
+
+# Regress into workdir and perform tests there, returning the test results.
+def autoregress(workdir, tmpdir, verbosity, clean):
+	logger = logging.getLogger(__name__)
+	dirname = os.path.basename(workdir)
+	logger.debug("In '%s'." % workdir)
+	for content in os.listdir(workdir):
+		content_full_path = os.path.join(workdir, content)
+		logger.debug("examining %s" % content_full_path)
+		if not os.path.isdir(content_full_path):
+			os.symlink(content_full_path, os.path.join(tmpdir,
+					content))
+	pre_files = os.listdir(tmpdir)
+	for content in os.listdir(tmpdir):
+		examine(os.path.join(tmpdir, content), verbosity)
+	post_files = os.listdir(tmpdir)
+	diff_files = list(set(post_files) - set(pre_files))
+	results = []
+	for diff_file in diff_files:
+		results.append(test(os.path.join(tmpdir, diff_file)))
+	if clean:
+		for thing in os.listdir(tmpdir):
+			if os.path.isdir(os.path.join(tmpdir, thing)):
+				shutil.rmtree(os.path.join(tmpdir, thing))
+			else:
+				os.remove(os.path.join(tmpdir, thing))
+	for content in os.listdir(workdir):
+		content_full_path = os.path.join(workdir, content)
+		if os.path.isdir(content_full_path) and not os.path.samefile(
+				tmpdir, content_full_path):
+			results.extend(autoregress(content_full_path, tmpdir,
+					verbosity, clean))
+	return results
+
+# Decide what to do to content and then do it.
+def examine(content, verbosity):
+	logger = logging.getLogger(__name__)
+	logger.debug("Calling examine() on %s" % content)
+	extension = os.path.splitext(content)[1]
+	if extension == ".cfg" or extension == ".config":
+		logger.debug("Calling asf_mapready on %s" % content)
+		dev_null = open(os.devnull, "w")
+		subprocess.call(["asf_mapready", content], stdout = dev_null,
```

Comment:
We shouldn't rely on the PATH being set such that the process will find just "asf_mapready". Instead, I'd suggest either putting the full path in here or in a config file.